src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -594,5 +594,23 @@ internal static Location FirstOrNone(this ImmutableArray<Location> items)
         {
             return items.IsEmpty ? Location.None : items[0];
         }
+
+        internal static bool SequenceEqual<TElement, TArg>(ImmutableArray<TElement> array1, ImmutableArray<TElement> array2, TArg arg, Func<TElement, TElement, TArg, bool> predicate)
```

Comment:
>TArg arg [](start = 125, length = 8)

Minor: It looks like other similar methods in this file put this parameter after the delegate that uses it.

Reply:
This ordering was intentional, and I'd argue that the rest of the functions in this file should change. By putting the argument before the lambda, we give the compiler enough info to infer the type of `TArg` before the lambda is typed, meaning that you get actual intellisense in the body.

---
In reply to: [327331056](https://github.com/dotnet/roslyn/pull/38769#discussion_r327331056) [](ancestors = 327331056)